src/ci/run.sh

```diff
@@ -134,6 +134,7 @@ if [ "$DEPLOY$DEPLOY_ALT" = "1" ]; then
 
   CODEGEN_BACKENDS="${CODEGEN_BACKENDS:-llvm}"
   RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set rust.codegen-backends=$CODEGEN_BACKENDS"
+  RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set build.compiletest-use-stage0-libtest=false"
```

Comment:
On most dist builders we probably don't even build compiletest, as they don't run tests. I would set this to `false` in the `dist` profile, in order to be enabled globally on CI, and then have one runner that sets it to `true`, to test that compiletest works also with stage0, if we want to gate on this.

Reply:
Not sure if using it in just one runner gives enough coverage.

Here's what I did: `compiletest-use-stage0-libtest` is set to `false` in the `dist` profile and `run.sh` script overrides it to `true` unless `$DEPLOY` or `$DEPLOY_ALT` is set.

Reply:
I'm pretty sure that most dist builders don't even build compiletest, as they don't run any tests. So if you want to have partial coverage of both true/false, we should do the split inside the test builders, not between test and dist, IMO.

Reply:
That being said, I'm not sure if we need that much coverage of the true value? It's just a hack to make local rebuilds faster. I would put it maybe into one runner per OS.

Although, if we have both true and false on CI as blocking, can't that break beta bumps? :thinking:

Reply:
> That being said, I'm not sure if we need that much coverage of the true value? It's just a hack to make local rebuilds faster. I would put it maybe into one runner per OS.

Yeah, probably... I can do that for Linux, Mac and Windows (let me know if I am missing any other) as those are the most popular ones used by devs.

> Although, if we have both true and false on CI as blocking, can't that break beta bumps? 🤔

`compiletest` is always built using the in-tree version of libtest. So if something breaks libtest, it would fail during development already; it wouldn't make it to beta without being caught.

Reply:
Hm yeah, I think there's a window where indeed `cfg(bootstrap)` might be needed, a library/testing-devex contributor who wants to modify libtest may need to `cfg(bootstrap)`-gate bit.

Reply:
That being said, following https://github.com/rust-lang/rust/pull/139317, the libtest usages in compiletest should be mostly isolated in one single module. We could add a doc comment on that about possibly needing `cfg(bootstrap)`-gating libtest changes.

Personally, I think this approach (possibility of compiletest using stage0 libtest for local dev) is reasonable, and I can't really think of a better solution.